packages/mui-material/src/styles/createThemeNoVars.js

```diff
@@ -24,10 +24,13 @@ function createThemeNoVars(options = {}, ...args) {
     ...other
   } = options;
 
-  if (options.vars) {
+  if (
+    options.vars &&
+    options.generateThemeVars === undefined // check for `generateThemeVars` (created internally) to handle nested theme
```

Comment:
Could you clarify in the comment why `options.generateThemeVars` being `undefined` is what we're looking for?